### PR TITLE
[detours] support debug builds

### PR DESCRIPTION
**Pull request by sylveon**
This enables to build Detours with a Debug config, which was recently added (https://github.com/microsoft/Detours/pull/175). Consumers not using `--head` still get benefits from this, namely custom triplets with different compiler flags in debug builds are respected.

- What does your PR fix? N/A

- Which triplets are supported/not supported? Have you updated the CI baseline? Yes

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/detours/CONTROL [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Source: detours
-Version: 4.0.1-1
+Version: 4.0.1-2
```
Comment (NancyLi1013) [inline ports/detours/CONTROL:2]:
```suggestion
Version: 4.0.1
Port-Version: 3
```
Please use `Port-Version` to express the changes for a port. See the doc here https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#port-version

File: ports/detours/CONTROL [hunk 2]
```diff
@@ -1,3 +1,3 @@
 Source: detours
-Version: 4.0.1-1
+Version: 4.0.1-2
 Description: Detours is a software package for monitoring and instrumenting API calls on Windows.
```
Comment (NancyLi1013) [inline ports/detours/CONTROL:3]:
```suggestion
Homepage: https://github.com/microsoft/Detours
Description: Detours is a software package for monitoring and instrumenting API calls on Windows.
```
Please also add `Homepage `here.

File: versions/baseline.json [hunk 3]
```diff
@@ -1597,7 +1597,7 @@
       "port-version": 1
     },
     "detours": {
-      "baseline": "4.0.1-1",
+      "baseline": "4.0.1-2",
```
Comment (NancyLi1013) [inline versions/baseline.json:1600]:
```suggestion
      "baseline": "4.0.1",
      "port-version": 3,
```

File: versions/d-/detours.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "f89604562d9880e41eaa00d36978f1c2b021179b",
+      "version-string": "4.0.1-2",
+      "port-version": 0
```
Comment (NancyLi1013) [inline versions/d-/detours.json:6]:
```suggestion
      "version-string": "4.0.1",
      "port-version": 3
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/detours/CONTROL (hunk 1 above)
Comment (sylveon) [reply ports/detours/CONTROL:2]: I tried using port versions but vcpkg complained it was a downgrade when I used vcpkg x-add-version

**Comment by NancyLi1013** [comment]
Hi @sylveon 

Thanks for your PR.

Please run `vcpkg x-add-version --overwrite-version detours` to update the .json files.

**Comment by sylveon** [comment]
Thanks, I changed it to use port versions and updated the baseline again.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Awesome, thanks for the upstream fix and thanks for the PR!

**Pull request merged**